### PR TITLE
[HGC trigger] V10/V11 e/g ID + calibration and concentrator module sums

### PR DESCRIPTION
**Pull request by jbsauvan**
#### PR description:
- Add e/gamma identification BDT and e/gamma energy corrections for the V10 and V11 geometries (@cerminar )
- Implement module sums of unselected trigger cells in the concentrator (@echapon )

Depends on external cms-data/L1Trigger-L1THGCal#18

#### PR validation:

Standard tests with different worflows have been run:
```
 runTheMatrix.py -w upgrade -l 20034.0 --maxSteps=2
 runTheMatrix.py -w upgrade -l 20434.0 --maxSteps=2
 runTheMatrix.py -w upgrade -l 22034.0 --maxSteps=2
 runTheMatrix.py -w upgrade -l 22434.0 --maxSteps=2
 runTheMatrix.py -w upgrade -l 27034.0 --maxSteps=2
```
Physics performance studies have also been done.

### CODE REVIEW

**Comment by silviodonato** [comment]
please test

**Comment by silviodonato** [comment]
abort

**Comment by silviodonato** [comment]
urgent

**Comment by silviodonato** [comment]
please test with cms-data/L1Trigger-L1THGCal#18

**Comment by kpedro88** [comment]
errors are in the IB, not caused by this PR

File: L1Trigger/L1THGCal/interface/concentrator/HGCalConcentratorTrigSumImpl.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#include "DataFormats/L1THGCal/interface/HGCalTriggerCell.h"
+#include "DataFormats/L1THGCal/interface/HGCalTriggerSums.h"
+#include "L1Trigger/L1THGCal/interface/HGCalTriggerTools.h"
+#include <vector>
+
+class HGCalConcentratorTrigSumImpl {
+public:
+  HGCalConcentratorTrigSumImpl(const edm::ParameterSet& conf);
+
+  void doSum(uint32_t module_id,
```
[the remark this answers is not in the document]
Comment (jbsauvan) [reply L1Trigger/L1THGCal/interface/concentrator/HGCalConcentratorTrigSumImpl.h:14]: Indeed it can be `const`. I'll change it.

**Comment by kpedro88** [comment]
test parameters
pull_request = cms-data/L1Trigger-L1THGCal#18

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
+upgrade

**Comment by silviodonato** [comment]
test parameters
pull_request =

**Comment by silviodonato** [comment]
test parameters

**Comment by silviodonato** [comment]
please test
let's test again after the integration of https://github.com/cms-sw/cmsdist/pull/5982
meanwhile @rekovic could you sign it?

**Comment by silviodonato** [comment]
@jbsauvan I'm going to merge this PR in the next IB. Please open a backport PR towards CMSSW_11_1_X in order to have this in CMSSW_11_1_0

**Comment by silviodonato** [comment]
please test with  cms-sw/cmsdist#5982

**Comment by silviodonato** [comment]
merge

**Comment by jbsauvan** [comment]
Thanks @silviodonato 
I just made the backport PR: #30355

**Comment by rekovic** [comment]
+1

**Pull request merged**